client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java

```diff
@@ -2263,4 +2263,19 @@ public boolean resumeCheckHalfMessage(final String addr, String msgId,
                 return false;
         }
     }
+
+    private static String join(List<String> list, String splitter) {
+        if (list == null) {
+            return null;
+        }
+        StringBuilder str = new StringBuilder();
+        for (int i = 0; i < list.size(); i++) {
+            str.append(list.get(i));
+            if (i == list.size() - 1) {
+                break;
+            }
+            str.append(splitter);
+        }
+        return str.toString();
+    }
```

Comment:
Maybe, you put this function into UtilAll or AclUtils class is better!

Reply:
Thanks for your advice, I have moved this two functions to UtilAll